code/modules/atmospherics/machinery/portable/pipe_scrubber.dm

```diff
@@ -0,0 +1,159 @@
+/obj/machinery/portable_atmospherics/pipe_scrubber
+	name = "pipe scrubber"
+	icon_state = "pipe_scrubber"
+	density = TRUE
+	max_integrity = 250
+	volume = 200
+	///The internal air tank obj of the mech
+	var/obj/machinery/portable_atmospherics/canister/internal_tank
+	///Is the machine on?
+	var/on = FALSE
+	///What direction is the machine pumping to (into scrubber or out to the port)?
+	var/direction = PUMP_IN
+	///the rate the machine will scrub air
+	var/volume_rate = 1000
+	///List of gases that can be scrubbed
+	var/list/scrubbing = list(
+		/datum/gas/plasma,
+		/datum/gas/carbon_dioxide,
+		/datum/gas/nitrous_oxide,
+		/datum/gas/bz,
+		/datum/gas/nitrium,
+		/datum/gas/tritium,
+		/datum/gas/hypernoblium,
+		/datum/gas/water_vapor,
+		/datum/gas/freon,
+		/datum/gas/hydrogen,
+		/datum/gas/healium,
+		/datum/gas/proto_nitrate,
+		/datum/gas/zauker,
+		/datum/gas/halon,
+	)
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/Initialize(mapload)
+	internal_tank = new(src)
+	. = ..()
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/Destroy()
+	var/turf/my_turf = get_turf(src)
+	my_turf.assume_air(air_contents)
+	my_turf.assume_air(internal_tank.air_contents)
+	SSair.stop_processing_machine(internal_tank)
+	qdel(internal_tank)
```

Comment:
what happens to the internal take variable when what it's pointed to is qdel'd? does it just hang onto a reference?

Reply:
I forgot how to subscribe to qdel of a child in byond, will add a handler.

Reply:
I don't mean we need to react to it, we essentially own that tank (unless it melts somehow that shouldn't happen right? is that your intention?) I mean if we destory the scrubber this'll hang a ref, I think you want QDEL_NULL